api-server/src/uploads/uploads.service.ts

```diff
@@ -25,28 +29,45 @@ export class UploadsService {
     }
   }
 
-  saveImage(img: Express.Multer.File) {
+  async saveImage(img: Express.Multer.File) {
+    const resizedImage = await this.resizeAndCompress(img);
+
     if (isNullOrUndefined(AWS_S3_UPLOADS_BUCKET)) {
       console.debug(
         'using local drive to persist image uploads, this should only be used for local testing',
       );
-      return this.saveToLocal(img);
+
+      return this.saveToLocal(resizedImage);
     } else {
-      return this.saveToS3(img);
+      return this.saveToS3(resizedImage);
     }
   }
 
-  private saveToLocal(img: Express.Multer.File): Promise<string> {
+  private async resizeAndCompress(
+    imgFile: Express.Multer.File,
+  ): Promise<Buffer> {
+    let imageInProcess: Sharp = sharp(imgFile.buffer);
+
+    const { width } = await imageInProcess.metadata();
+
+    if (width > IMAGE_DESIRED_WIDTH) {
+      imageInProcess = imageInProcess.resize(IMAGE_DESIRED_WIDTH);
+    }
+
+    return imageInProcess.webp().toBuffer();
+  }
+
+  private saveToLocal(imgBuffer: Buffer): Promise<string> {
     const subPath = 'event-images';
 
     return new Promise((resolve, reject) => {
-      const imageName = this.generateNewImageName(img);
+      const imageName = this.generateNewImageName();
 
       // wx flag mitigates the possibility of clobbering an existing file
       // (will error out on write)
       fs.writeFile(
         join(PATH_TO_LOCAL_EVENT_IMAGE_UPLOADS, imageName),
```

Comment:
Is this path in `.gitignore`?

Reply:
PATH_TO_LOCAL_EVENT_IMAGE_UPLOADS?

I believe so, though it's kind of weird because depending on how things are ran it  often actually pops up under dist (also in .gitignore, I had a fair bit of trouble with this actually).

That specific comment is probably copy/paste from the old code in the nuxt probject, but still looks true. This is all for local test, so nbd